jsk_fetch_robot/jsk_fetch_startup/launch/rosbag_record.launch

```diff
@@ -30,7 +33,10 @@
           /head_camera/depth_registered/throttled/camera_info
           /head_camera/rgb/throttled/image_rect_color/compressed
           /head_camera/depth_registered/throttled/image_rect/compressedDepth
-          /audio"
+          /audio
+          $(arg other_topics)
+          -e $(arg regex_topics)
```

Comment:
when `arg regex_topics` is empty, this launch may raise error.
We should have `use_regex` flag or `eval regex_topics != ""` to avoid the error.

Reply:
@708yamaguchi does this work properly with empty string in`regex_topics`.
What happens when `regex_topics` is empty and `other_options` has some strings?

Reply:
can you check the same command in command line, not in roslaunch?
it seems rosbag does not uses `rospy.myargv`, so it passes all the arguments for roslaunc process, I think.
So my guess is that you will get error when you run `rosbag record -O out.bag /topics -e` in command line.

Reply:
I fixed it. If there is no problem, I will push the same commit to pr2.

Reply:
> So my guess is that you will get error when you run rosbag record -O out.bag /topics -e in command line.

I did not get errors...

```
fetch@fetch1075:~$ rosbag record -O out.bag /tf -e
[ INFO] [WallTime: 1654345481.431464434] [node:/record_1654345481383153471] [func:Recorder::startWriting]: Recording to 'out.bag'.
[ INFO] [WallTime: 1654345481.436231980] [node:/record_1654345481383153471] [func:shared_ptr<ros::Subscriber> rosbag::Recorder::subscribe]: Subscribing to /tf
```

```
fetch@fetch1075:~$ rosbag record -O out.bag /tf -e -l 10000
[ INFO] [WallTime: 1654345633.860405687] [node:/record_1654345633787606342] [func:Recorder::startWriting]: Recording to 'out.bag'.
[ INFO] [WallTime: 1654345633.865821367] [node:/record_1654345633787606342] [func:shared_ptr<ros::Subscriber> rosbag::Recorder::subscribe]: Subscribing to /tf
```

> it seems rosbag does not uses rospy.myargv

`rosbag record` uses `Boost Program Options` to parse arguments, but I don't know its behavior.
https://github.com/ros/ros_comm/blob/dd78ac8af128bb8eb992d6431bb9f994658ea6ab/tools/rosbag/src/record.cpp

Reply:
@nakane11 

> I fixed it. If there is no problem, I will push the same commit to pr2.

Using `eval` is good solution, but it is available in kinetic or newer.
https://wiki.ros.org/roslaunch/XML#substitution_args

So I think this solution is good for fetch (melodic), but may not be good for PR2(indigo).

Reply:
First it will run rosbag in python and it will call c++ program in subprocess.
Then it is a bit weird behavior. I will check it later.

Reply:
> First it will run rosbag in python and it will call c++ program in subprocess.
> Then it is a bit weird behavior. I will check it later.

I see, thank you.

Reply:
@708yamaguchi 
Should I avoid `eval` to unify arguments?

Reply:
https://github.com/jsk-ros-pkg/jsk_common/blob/master/jsk_data/launch/common_record.launch
https://github.com/jsk-ros-pkg/jsk_common/blob/master/jsk_data/launch/fetch_record.launch
I think these launch files have been working for many years.

Reply:
I searched a bit about `--regex, -e` flag, and I found the correct usage.
`--regex` is just store the `True` value when the flag is there.
So I think you can just `-e` flag and not need to write `regex topics` after the flag.

However , `-x EXCLUDE_REGEX, --exclude=EXCLUDE=REGEX` is different.
If you want to use `-x`, you need to write `exclud regrex topic` just after the flag.

So my conclusion is that you don't need to separate `other_topics` and `regex_topic`, just merge it.
And `--regex` is just a flag, so you can add like this.

```xml
  <arg name="regex_flag" value="--regex" if="$(arg use_regex)" />
  <arg name="regex_flag" value="" unless="$(arg use_regex)" />
...
...
          /audio
          $(arg other_topics)
          $(arg regex_flag) $(arg other_options)
```
http://wiki.ros.org/rosbag/Commandline#record

Reply:
@knorth55 
Thank you. I checked `other_topics:=regex_topicname use_regex:=true` works and `other_topics:=regex_topicname`  does not work.
I wonder if I should set `use_regex=true` by default.

Reply:
i think you can set `use_regex` `true` as default.